### ISSUE TITLE
test_int.test_instance_set failing on .NET 7

### ISSUE THREAD

**Issue by slozier** [IronLanguages/ironpython3#1485]
.NET 7 adds a number of new methods and properties to `Int32` which are not available on `BigInteger`: `TrailingZeroCount`, `Log2`, `MinMagnitude`, `PopCount`, `TryCreate`, `CreateSaturating`, `CreateTruncating`, `LeadingZeroCount`, `MaxMagnitude`, `CreateChecked`, `WriteLittleEndian`, `IsPow2`, `CopySign`, `IsNegative`, `Clamp`, `TryWriteLittleEndian`, `RotateLeft`, `RotateRight`, `GetShortestBitLength`.

@BCSharp I guess we don't care about `Int32` instances having additional members and we can just relax the assertion?

Related to https://github.com/IronLanguages/ironpython3/pull/1399.

**Comment by BCSharp** [comment]
> I guess we don't care about `Int32` instances having additional members and we can just relax the assertion?

Not really. The assertion was mainly to provide visibility into what `Int32` defines extra and ring the bell if something changes in a new .NET version to decide if something is worth implementing. For instance, `BigInteger` has `IsPowerOfTwo` (which `Int32Ops` mimics) and now `Int32` is supposed to have `IsPow2`? What the heck? By the way, where did you find those new methods/properties documented?

**Comment by slozier** [comment]
> now Int32 is supposed to have IsPow2?

Looks like it's been discussed during API review and they went with `IsPow2`... https://github.com/dotnet/runtime/issues/31297#issuecomment-644560126

> By the way, where did you find those new methods/properties documented?

I just ran the test with .NET 7 and it spewed out those as missing attributes. I'm guessing they're all part of the new numeric interfaces (e.g. [System.IBinaryNumber](https://docs.microsoft.com/en-us/dotnet/api/system.ibinarynumber-1?view=net-7.0)).

**Comment by BCSharp** [comment]
Thanks. Hopefully `BigInteger` implements `System.IBinaryNumber` (and why not) and this particular issue will solve itself.

**Comment by slozier** [comment]
Ah, looks like they did add `System.IBinaryNumber` to `BigInteger` so maybe this issue will disappear in the next preview release!

**Comment by slozier** [comment]
A related failure which might not be resolved is that they added a static `Sign` method on `Int32` so `(1).Sign` now returns the method instead of calling the `IntOps` extension...
